_source/_posts/2022-07-21-jhipster-ci-cd.md

```diff
@@ -0,0 +1,867 @@
+
+```shell
+CONTEXT=$(kubectl config current-context)
+
+kubectl apply --context $CONTEXT \
+    -f ./jhipster-service-account.yml
+
+
+TOKEN=$(kubectl get secret --context $CONTEXT \
+   $(kubectl get serviceaccount jhipster-service-account \
+       --context $CONTEXT \
+       -n demo \
+       -o jsonpath='{.secrets[0].name}') \
+   -n demo \
+   -o jsonpath='{.data.token}' | base64 --decode)
+
+kubectl config set-credentials ${CONTEXT}-token-user --token $TOKEN
+
+kubectl config set-context $CONTEXT --user ${CONTEXT}-token-user
+```
+
+Add the account to the Halyard configuration:
+
+```shell
+CONTEXT=$(kubectl config current-context)
+
+hal config provider kubernetes account add jhipster-gke-account \
+    --context $CONTEXT
+```
+
+### Add a Docker registry account
+
+As the goal is to trigger the pipeline execution when a new image is pushed to DockerHub, you need to configure a docker-registry provider with Halyard.
+
+First, enable the docker-registry provider:
+
+```shell
+hal config provider docker-registry enable
+```
+
+The following `hal config` line will prompt for your password or access token if you have 2FA enabled. Make sure to replace `your-dockerhub-username`.
+
+```shell
+ADDRESS=index.docker.io
+USERNAME=your-dockerhub-username
+
+hal config provider docker-registry account add docker-account \
+    --address $ADDRESS \
+    --repositories $USERNAME/store $USERNAME/gateway \
+    --username $USERNAME \
+    --password
+```
+
+### Add a GitHub artifact account
+
+The pipeline will deploy k8s manifests from a GitHub repository, so you must also configure a GitHub artifact account. First, enable the feature:
+
+```shell
+hal config artifact github enable
+```
+Generate an access token on GitHub. Sign in, and in the user menu choose **Settings**. Then on the left menu choose **Developer settings**. On the left menu choose **Personal access tokens**. Click **Generate new token** and copy the token.
+
+Then add your GitHub username and set the new token when prompted:
+
+```shell
+ARTIFACT_ACCOUNT_NAME=your-github-username
+
+hal config artifact github account add $ARTIFACT_ACCOUNT_NAME \
+    --token
+```
+
+The new GKE, Docker, and GitHub account configurations must be applied to the deployment before starting the pipeline design:
+
+```shell
+hal deploy apply
+```
+
+### Create the store microservice pipeline
+
+The _pipeline_ is the central concept in deployment management with Spinnaker. It is compounded by a sequence of actions, named _stages_. A pipeline can be triggered manually or automatically, and have an execution history. Connect to the Spinnaker UI again to create the first pipeline.
+
+```shell
+hal deploy connect
+```
+
+Navigate to `https://localhost:9000`, create the store application, and choose Kubernetes as the cloud provider. Click **Create Application** and set the following values:
+
+{% img blog/jhipster-ci-cd/sp-store-app.png alt:"Spinnaker create application" width:"550" %}{: .center-image }
```

Comment:
You only show the name filled in here. It's obvious I need to add my email, but should I also select GitHub and enter my repo information? If so, please state that.

Reply:
Added a comment.